packages/@lwc/ssr-compiler/src/compile-js/generate-markup.ts

```diff
@@ -30,6 +31,7 @@ type RenderCallExpression = SimpleCallExpression & {
 
 const bGenerateMarkup = esTemplate`
     export async function* generateMarkup(tagName, props, attrs, slotted, parent, scopeToken) {
+        tagName = tagName ?? ${/*component tag name*/ is.literal}
```

Comment:
Sample output will look like this:

```js
async function* generateMarkup$1(tagName, props, attrs, slotted, parent, scopeToken) {
  tagName = tagName ?? "x-test";
```
We need this to get the tag name that's passed to the transformer, otherwise, the constructor doesn't have a tag name to use.